buf/registry/module/v1beta1/branch_service.proto

```diff
@@ -98,3 +118,117 @@ message ListBranchesResponse {
   // The listed Branches.
   repeated Branch branches = 2;
 }
+
+message PushCommitsRequest {
+  // A request to push a Commit to a Branch.
+  message Value {
+    // The Branch to push the Commit to.
+    BranchRef branch_ref = 1 [(buf.validate.field).required = true];
+    // The ID of the Commit to push to the Branch.
+    //
+    // This Commit will become the latest Commit on the Branch.
+    //
+    // If the Commit already exists on the Branch, an error is returned.
+    string commit_id = 2 [
```

Comment:
`Commit`s don't have a `name`, I think that's why there's no `CommitRef`. That leaves two ways to reference a `Commit`: `id` or `digest`. We use `id` everywhere else (`Tag`, `Branch`, `VCSCommit`), so I stuck with that.